src/python/grpcio/grpc/experimental/__init__.py

```diff
@@ -30,3 +35,40 @@ class ChannelOptions(object):
 
 class UsageError(Exception):
     """Raised by the gRPC library to indicate usage not allowed by the API."""
+
+
+_insecure_channel_credentials = object()
+
+
+def insecure_channel_credentials():
+    """Creates a ChannelCredentials for use with an insecure channel.
+
+    THIS IS AN EXPERIMENTAL API.
+
+    This is not for use with secure_channel function. Intead, this should be
+    used with grpc.unary_unary, grpc.unary_stream, grpc.stream_unary, or
+    grpc.stream_stream.
+    """
+    return grpc.ChannelCredentials(_insecure_channel_credentials)
+
+
+class ExperimentalApiWarning(Warning):
+    """A warning that an API is experimental."""
+
+
+def warn_experimental(api_name):
+    msg = ("{} is an experimental API. It is subject to change or ".format(
+        api_name) + "removal between minor releases. Proceed with caution.")
+    warnings.warn(msg, ExperimentalApiWarning, stacklevel=2)
```

Comment:
optional: should we make it a decorator? so we can skip typing `api_name` explicitly.

Reply:
I wanted to be able to use this for arbitrary objects. Not just functions, but also classes. A decorator needs to know what sort of object it's decorating. Making it a parameter gives the author of the API the ability to determine the exact scope of what they are calling an experimental API. Thoughts?

Reply:
Discussed offline. Just removing the warnings since inclusion in the `experimental` module should be enough to signal to potential users that there is no guarantee of API stability here.